gui/controller_history.h

```diff
@@ -0,0 +1,41 @@
+#ifndef CONTROLLER_HISTORY_H
+#define CONTROLLER_HISTORY_H
+
+#include "../common/generated_libs/network_protocol/network_protocol.pb.h"
+
+#include <deque>
+
+// Maintains a history of recent ControllerStatus-es sufficient
+// for rendering the UI.
+//
+// Non-thread-safe, needs external synchronization.
+class ControllerHistory {
+public:
+  using Instant = std::chrono::time_point<std::chrono::steady_clock>;
+
+  ControllerHistory(std::chrono::milliseconds window) : window_(window) {}
+
+  void Append(Instant gui_now, const ControllerStatus &status) {
```

Comment:
Perhaps as a TODO, I feel like there should be some protection from the controller sending a gazillion statuses in a row and OOMing the GUI.

Using gui_now instead of the controller's timestamp is significant here.  Since the controller doesn't have an absolute clock, just ms-since-startup (at least, that's how the Arduino is), we *have* to do it this way if we want to survive controller restarts.  I think worth a comment; I would have written this code wrong.

Reply:
Yeah, this is exactly why I used a steady clock here - to protect against controller restart. Will add comments.

Reply:
Added.